src/map.cpp

```diff
@@ -904,14 +904,20 @@ void map::unboard_vehicle( const tripoint &p )
         return;
     }
     passenger = vp->vehicle().get_passenger( vp->part_index() );
+    // Mark the part as un-occupied regardless of whether there's a live passenger here.
+    vp->part().remove_flag( vehicle_part::passenger_flag );
     if( !passenger ) {
-        debugmsg( "map::unboard_vehicle: passenger not found" );
+        if( !dead_passenger ) {
```

Comment:
Does this really need a flag for a single `debugmsg`? I mean, why not bail out silently whenever there's no passenger found?

Reply:
Because if we expect there to be a passenger there, it's a big deal that we were wrong about it.